tests/testthat/test-expr_string.R

```diff
@@ -604,20 +604,30 @@ test_that("str$replace", {
   )
 
   expect_identical(
-    pl$lit(c("123abc", "abc456"))$str$replace(r"{abc\b}", "ABC", TRUE)$to_r(),
+    pl$lit(c("123abc", "abc456"))$str$replace(r"{abc\b}", "ABC", literal = TRUE)$to_r(),
     c("123abc", "abc456")
   )
 
   e = pl$lit(r"{(abc\b)}")
   expect_identical(
-    pl$lit(c("123abc", "abc456"))$str$replace(e, "ABC", FALSE)$to_r(),
+    pl$lit(c("123abc", "abc456"))$str$replace(e, "ABC", literal = FALSE)$to_r(),
     c("123ABC", "abc456")
   )
 
   expect_identical(
     pl$lit(c("abcabc", "123a123"))$str$replace("ab", "__")$to_r(),
     c("__cabc", "123a123")
   )
+
+  expect_identical(
+    pl$lit(c("ababab", "123a123"))$str$replace("a", "_", n = 2)$to_r(),
+    c("_b_bab", "123_123")
+  )
+
+  expect_error(
```

Comment:
To check but I'm not sure `expect_error()` works correctly. Can you try with a fake expected regex? Maybe `expect_grepl_error()` should be used instead

Reply:
I do not know what your concern is. I just want to confirm here that the combination of regular expression and `n>2` will not work and will result in an error.

Also it is not clear to me if `expect_grepl_error()` is better than `expect_error()`. (I don't know why `expect_grepl_error()` exists here)
As far as I know, `expect_error()` supports capturing error messages with regular expressions.

Reply:
> As far as I know, `expect_error()` supports capturing error messages with regular expressions.

The structure of polars errors doesn't seem to work well with regexes in `expect_error()`:

```r
# This errors (as it should)
pl$lit("?")$str$decode("base64")$to_r()

# This passes, even though "foobar" is not in the error message
expect_error(
    pl$lit("?")$str$decode("base64")$to_r(),
    "foobar"
  )

# This doesn't pass, as expected
  expect_grepl_error(
    pl$lit("?")$str$decode("base64")$to_r(),
    "foobar"
  )

Error: expected_err[which(!founds)[1]] (`actual`) not identical to `err` (`expected`).

    lines(actual) | lines(expected)                                                                    
[1] "foobar"      - "Error: Execution halted with the following contexts"                           [1]
                  - "   0: In R: in $select()"                                                      [2]
                  - "   1: Encountered the following error in Rust-Polars:"                         [3]
                  - "      \tinvalid `base64` encoding found; try setting `strict=false` to ignore" [4]
                  - ""                                                                              [5]
                  - ""                                                                              [6]

Reply:
I see, thanks.